streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -1783,10 +1783,39 @@ void to(final Serde<K> keySerde,
                                      final ValueJoiner<? super V, ? super VO, ? extends VR> joiner,
                                      final StateStoreSupplier<KeyValueStore> storeSupplier);
 
+    
+    
+    /**
+     * 
+     * Joins one record of this KTable to n records of the other KTable
+     * an event in this KTabl will produce n consecutive events an update
+     * in other table will update only the 1 matched record
+     * 
+     * @param the table containing n records for each K of this table
+     * @param keyExtractor a {@code ValueMapper} returning the key of this table from the others value
+     * @param joinPrefixFaker a {@code ValueMapper} returning an outputkey that when serialized only produces the
+     * 							prefix of the output key wich is the same as serializing K 
+     * @param leftKeyExtractor a {@code ValueMapper} extracting the Key of this table from the resulting Key 
+     * @param <K0> the resultings tables Key
+     * @param <V0> the resultings tables Value
+     * @param joiner
+     * @return
+     */
+    <K0, V0, KO, VO> KTable<K0, V0> oneToManyJoin(KTable<KO, VO> other,
```

Comment:
returns new K0 wich needs to contain K and K0. I love how this works so i expect controversy here aswell